### PR TITLE
Fixed/Checked ALL the loans features to confirm it works after refactoring loans-ms to accounts-ms

### PR DESCRIPTION
**Pull request by cpatelss90**

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,6 +1,44 @@
...
+
+
+
+
+
+#spring.datasource.url=jdbc:mysql://database-1.ctppvz686u4m.us-east-2.rds.amazonaws.com:3306/utopia_cashmoney?useSSL=false
+#spring.datasource.username=admin
+#spring.datasource.password=password
+
+spring.datasource.url=jdbc:mysql://localhost:3306/utopia_cashmoney?useSSL=false
```
Comment (JostenAsercion) [inline src/main/resources/application.properties:39]: We definitely should not be going back to local databases instead of RDS

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,6 +1,44 @@
-spring.datasource.url=jdbc:mysql://database-1.ctppvz686u4m.us-east-2.rds.amazonaws.com:3306/utopia_cashmoney?useSSL=false
-spring.datasource.username=admin
-spring.datasource.password=password
+#spring.datasource.url=jdbc:mysql://database-1.ctppvz686u4m.us-east-2.rds.amazonaws.com:3306/utopia_cashmoney?useSSL=false
+#spring.datasource.username=admin
+#spring.datasource.password=password
+#
+#spring.datasource.driver-class-name =com.mysql.cj.jdbc.Driver
+#spring.jpa.hibernate.ddl-auto=validate
+#server.port=8081
```
Comment (JostenAsercion) [inline src/main/resources/application.properties:7]: I'm really not sure why this is commented out. Having the /api context path is very important to the functionality of the microservice. This may fix it for how you implemented things on the user ui side, but this will break everything else. Opt to add /api to your backend calls instead. Also why comment out the port?

File: src/main/java/com/ucm/ms/accounts/controllers/CreditCardController.java [hunk 3]
```diff
@@ -37,4 +67,119 @@ public CreditCardController(CreditCardSearch creditCardSearch) {
 			return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
 		}
 	}
-}
+	
+	
+		
+		
+		@GetMapping(path = "get_credit_cards")
```
Comment (JostenAsercion) [inline src/main/java/com/ucm/ms/accounts/controllers/CreditCardController.java:74]: We already have a method for getting all credit cards. What is this exactly? I notice that it is of type AccountType too.

File: src/main/java/com/ucm/ms/accounts/controllers/CreditCardController.java [hunk 4]
```diff
@@ -37,4 +67,119 @@ public CreditCardController(CreditCardSearch creditCardSearch) {
...
+				creditCard.setExpirationDate(creditCardExpiryDate);
+				
+				
+				userAccountDao.save(userAccount);
+				creditCardDao.save(creditCard);	
+				
+		}
+		
+		
+		@GetMapping("/user_account")
```
Comment (JostenAsercion) [inline src/main/java/com/ucm/ms/accounts/controllers/CreditCardController.java:164]: Why is this here? We have a userAccountsController. You can make adjustments in that file to take in the userId as a parameter there if you want.

File: src/main/java/com/ucm/ms/accounts/controllers/CreditCardController.java [hunk 5]
```diff
@@ -37,4 +67,119 @@ public CreditCardController(CreditCardSearch creditCardSearch) {
...
+
+			} catch (Exception e) {
+				return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+			}
+
+		}
+		
+		
+		@PostMapping(path = "/user_credit_card_signup")
+		public void usercreditCardSignUp(@RequestParam int userId, @RequestParam String cardName) {
```
Comment (JostenAsercion) [inline src/main/java/com/ucm/ms/accounts/controllers/CreditCardController.java:92]: This should definitely be part of a service instead of in the controller

File: src/main/java/com/ucm/ms/accounts/controllers/LoansController.java [hunk 6]
```diff
@@ -332,6 +340,72 @@ public void createLoans(@RequestBody RequestAccountTypeDto accountTypeDto) {
...
 			
 			
+		} catch (Exception e) {
+			return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+	}
+	
+	
+	@PostMapping("/submit_payment")
+	public void createLoans(@RequestParam int userId, @RequestParam double amount, @RequestParam String payeeAccountNumber) {
```
Comment (JostenAsercion) [inline src/main/java/com/ucm/ms/accounts/controllers/LoansController.java:350]: Again, this should definitely be a service and not in the controller

**Review by JostenAsercion — CHANGES_REQUESTED** [review]
I'm a bit confused on some of the changes here, let me know when you get back to this and I can review it again

File: src/main/resources/application.properties (hunk 1 above)
Comment (cpatelss90) [reply src/main/resources/application.properties:39]: I am using mysql locally but we can keep the rds for developer branch

File: src/main/resources/application.properties (hunk 2 above)
Comment (cpatelss90) [reply src/main/resources/application.properties:7]: oh that was just commented out but we do have all those parameters if you scroll down the file

File: src/main/java/com/ucm/ms/accounts/controllers/CreditCardController.java (hunk 3 above)
Comment (cpatelss90) [reply src/main/java/com/ucm/ms/accounts/controllers/CreditCardController.java:74]: The one we already have is from before and it does not have type AccountType. Its of type CreditCard. So I guess we can take down that old method and use this one instead since this is updated one.

Comment (JostenAsercion) [reply src/main/java/com/ucm/ms/accounts/controllers/CreditCardController.java:74]: Oh no that's not an old one, I made that method after we made the database changes. I think there might be some miscommunication here; are these meant to be returning different types of credit cards? As in like different account types that are of type credit card? The method that I made gets every individual credit card in the database. I'm thinking this one has a different purpose, so we should probably rename it.

File: src/main/resources/application.properties (hunk 2 above)
Comment (JostenAsercion) [reply src/main/resources/application.properties:7]: Ah, yeah I see that now my bad

**Review by JostenAsercion — COMMENTED** [review]
added a few comments

File: src/main/java/com/ucm/ms/accounts/controllers/CreditCardController.java (hunk 3 above)
Comment (cpatelss90) [reply src/main/java/com/ucm/ms/accounts/controllers/CreditCardController.java:74]: Oh yeah thats prolly it, I will rename it

**Comment by cpatelss90** [comment]
I have made a new PR with all the issues from this pr resolved.
